zebra-chain/src/primitives/zcash_note_encryption.rs

```diff
@@ -0,0 +1,68 @@
+//! Contains code that interfaces with the zcash_note_encryption crate from
+//! librustzcash.
+//!
+use crate::{
+    block::Height,
+    parameters::{Network, NetworkUpgrade},
+    primitives::zcash_primitives::convert_tx_to_librustzcash,
+    transaction::Transaction,
+};
+
+/// Returns if all Sapling or Orchard outputs, if any, decrypt successfully with
+/// an all-zeroes outgoing viewing key.
+///
+/// # Panics
+///
+/// If passed a network/height without matching consensus branch ID (pre-Overwinter),
+/// since `librustzcash` won't be able to parse it.
+pub fn decrypts_successfully(transaction: &Transaction, network: Network, height: Height) -> bool {
+    let network_upgrade = NetworkUpgrade::current(network, height);
+    let alt_tx = convert_tx_to_librustzcash(transaction, network_upgrade)
+        .expect("zcash_primitives and Zebra transaction formats must be compatible");
+
+    let alt_height = height.0.into();
+    let null_sapling_ovk = zcash_primitives::sapling::keys::OutgoingViewingKey([0u8; 32]);
+    if let Some(bundle) = alt_tx.sapling_bundle() {
+        for output in bundle.shielded_outputs.iter() {
+            let r = match network {
```

Comment:
```suggestion
            let recovery = match network {
```
Maybe it can be a bit more clear if we change this `r` name to `recovery`.